components/pages/task/create-task/controller.tsx

```diff
@@ -468,6 +469,13 @@ export default function CreateTaskPage({
     }
   }
 
+  useEffect(() => {
+    const marketplace = allNetworks?.find(m => lowerCaseCompare(m?.name, query?.marketplace?.toString()));
+    if (!query?.marketplace || !marketplace)
```

Comment:
this condition should happen before we loop, so we can avoid if there's nothing to do:

```ts
if (!query?.marketplace)
  return;

const marketplace = allNetworks.find(m => lowercaseCompare(m.name, query.marketplace.toString());

if (!marketplace?.chain)
  return;

handleNetworkSelected(marketplace.chain)

```